src/gtk/textctrl.cpp

```diff
@@ -560,6 +560,25 @@ au_insert_text_callback(GtkTextBuffer *buffer,
         wxGtkTextApplyTagsFromAttr(win->GetHandle(), buffer, win->GetDefaultStyle(),
                                    &start, end);
     }
+    const auto maxlen = win->GTKGetMaxLength();
+    auto count = gtk_text_buffer_get_char_count( buffer );
+    if ( maxlen > 0 && count > maxlen )
```

Comment:
It's pointless to call `gtk_text_buffer_get_char_count()` if `maxlen` is not set, so this should be:

```suggestion
    if ( maxlen > 0 )
    {
        const auto count = gtk_text_buffer_get_char_count( buffer );
        if ( count > maxlen )
```

(maybe this function is cheap to call, but maybe it isn't, and it's simpler to not to even have to think about it by avoiding calling it unless necessary).